design-proposals/vm-memory-dump.md

```diff
@@ -0,0 +1,147 @@
+# Overview
+Proposal for taking VM memory dump for analysis purposes.
+
+## Motivation
+One of the methods of troubleshooting guest virtual machines is getting a memory dump and analyzing it. As this is a standard capability in most VM management systems, it is also required for kubevirt’s VMs.
+
+## Goals
+Have a mechanism to get a VM memory dump.
+
+## Non Goals
+Memory dump is not a restorable output and currently will not be a part of the VM snapshot.
+
+## User Stories
+As a Kubevirt user I would like to get a memory dump of a running VM so that I can later inspect its memory.
+
+## Repos
+[KubeVirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+To trigger a memory dump a virtctl memoryDump command will be added.
+The command will either use an existing PVC or create a new PVC on demand. This PVC will be bound to the VM and will appear in the VM volumes spec with a fitting status.
```

Comment:
If memory dump to an existing PVC, we need create the PVC before starting the dump, right? if yes, how to know the size for the PVC that will be enough for dump?

Reply:
right, there is a formula to calculate the size of the memory used by a vm (example of the calculation done in oVirt: https://github.com/oVirt/ovirt-engine/blob/ovirt-engine-4.4.10.6/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/utils/VmOverheadCalculatorImpl.java#L135)